pyk/src/pyk/cfg_manager.py

```diff
@@ -0,0 +1,247 @@
+    bottom_up,
+    collectFreeVars,
+    count_vars,
+    extract_lhs,
+    extract_rhs,
+    flatten_label,
+    ml_pred_to_bool,
+    remove_generated_cells,
+    splitConfigFrom,
+    substitute,
+)
+from .kcfg import KCFG
+from .prelude import Bool, mlAnd, mlBottom, mlTop
+
+_LOGGER: Final = logging.getLogger(__name__)
+
+
+def instantiate_cell_vars(definition: KDefinition, term: KInner) -> KInner:
+    def _cellVarsToLabels(_kast):
+        if type(_kast) is KApply and _kast.is_cell:
+            production = definition.production_for_klabel(_kast.label)
+            production_arity = [prod_item.sort for prod_item in production.items if type(prod_item) is KNonTerminal]
+            new_args = []
+            for sort, arg in zip(production_arity, _kast.args):
+                if sort.name.endswith('Cell') and type(arg) is KVariable:
+                    new_args.append(definition.empty_config(sort))
+                else:
+                    new_args.append(arg)
+            return KApply(_kast.label, new_args)
+        return _kast
+    return bottom_up(_cellVarsToLabels, term)
+
+
+def rename_generated_vars(cterm: CTerm) -> CTerm:
+    state, *constraints = cterm
+    _, config_subst = splitConfigFrom(state)
+    config_var_count = {cvar: count_vars(ccontents) for cvar, ccontents in config_subst.items()}
+    free_vars = collectFreeVars(cterm.term)
+    var_subst: Dict[str, KInner] = {}
+    for v in free_vars:
+        if v.startswith('_Gen') or v.startswith('?_Gen') or v.startswith('_DotVar') or v.startswith('?_DotVar'):
+            cvars = [cv for cv in config_var_count if v in config_var_count[cv]]
+            if len(cvars) > 1:
+                raise ValueError(f'Found "Gen*" or "DotVar*" variable with multiple occurances: {v}')
+            cvar = cvars[0]
+            new_v = abstract_term_safely(KVariable(v), base_name=cvar)
+            while new_v.name in free_vars:
+                new_v = abstract_term_safely(KVariable(new_v.name), base_name=cvar)
+            var_subst[v] = new_v
+            free_vars.append(new_v.name)
+    return CTerm(Subst(var_subst).apply(cterm.term))
+
+
+def sanitize_config(defn: KDefinition, init_term: KInner) -> KInner:
+
+    def _var_name(vname):
+        new_vname = vname
+        while new_vname.startswith('_') or new_vname.startswith('?'):
+            new_vname = new_vname[1:]
+        return new_vname
+
+    free_vars_subst = {vname: KVariable(_var_name(vname)) for vname in collectFreeVars(init_term)}
+
+    # TODO: This is somewhat hacky. We shouldn't have to touch the config this much.
+    # Likely, the frontend should just be giving us initial states with these already in place.
+    def _remove_cell_map_definedness(_kast):
+        if type(_kast) is KApply:
+            if _kast.label.name.endswith('CellMap:in_keys'):
+                return Bool.false
+            elif _kast.label.name.endswith('CellMapItem'):
+                return _kast.args[1]
+        return _kast
```

Comment:
Remove this block entirely, we can re-implement downstream and find a better solution.

Reply:
Removed